src/UWP/UwpApiConnector.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\UWP;
+
+use CultuurNet\UDB3\Json;
+use CultuurNet\UDB3\Model\ValueObject\Identity\Uuid;
+use CultuurNet\UDB3\UWP\Exception\UwpApiFailure;
+use CultuurNet\UDB3\UWP\Result\VerenigingsloketConnectionResult;
+use Exception;
+use GuzzleHttp\Psr7\Request;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+
+class UwpApiConnector
+{
+    private ?string $accessToken = null;
+
+    public function __construct(
+        private ClientInterface $httpClient,
+        private string $clientId,
+        private string $clientSecret,
+        private LoggerInterface $logger
+    ) {
+    }
+
+    private function fetchAccessToken(): string
+    {
+        if ($this->accessToken !== null) {
+            return $this->accessToken;
+        }
+
+        $request = new Request(
+            'POST',
+            'https://auth.uitwisselingsplatform.be/realms/ddt/protocol/openid-connect/token',
```

Comment:
Should this not be part of a config?

Reply:
I though about it, started witht hat, but given that it is the same everywhere on every env, I did not see the benefit, it is also not private info like a client id or secret.